packages/gatsby/src/utils/webpack-utils.js

```diff
@@ -370,7 +370,12 @@ module.exports = async ({
               return true
             }
             // If dep is babel-runtime or core-js, exclude
-            if (/@babel(?:\/|\\{1,2})runtime|core-js/.test(modulePath)) {
+            // TODO this needs rework, this is buggy as hell
+            if (
+              /@babel(?:\/|\\{1,2})runtime|core-js|react|react-dom|scheduler|prop-types/.test(
```

Comment:
this regex probably needs directory boundaries to avoid `/react/` matching `react-foo`.